### ISSUE TITLE
Check if 'overlayfs' can be used by default

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/release-plan#8]
* Netrunner reported some issues with **overlayfs**
* we have to check if we have any issues by using **overlayfs**

**Comment by Kirek** [comment]
They do? I thought the still used and old version of manjaro-tools because they didn't convert the profiles.
In my end the only issue I have is the "machine-id" warning in the livecd.

**Comment by philmmanjaro** [comment]
Netrunner adopted to the newest version of **manjaro-tools**. My last info was, that they still have to use **aufs**.

**Comment by udeved** [comment]
We need to sort out machine-id.
@Kirek 

The warning my be caused because systemd needs it somehow earlier than livecd script is run?

Its no problem with openrc, since dbis script would create it anyway.

**Comment by Kirek** [comment]
Relevant bug report in systemd https://github.com/systemd/systemd/issues/729
TL;DR: The problem is in overlayfs, the warning is not a big deal: debian/ubuntu dropped a patch for it they didn't care enough and feels like a workaround).
And maybe linux 4.2 fixed it (need to check...).

**Comment by udeved** [comment]
I see, I mentioned it on forum.
I thought is calamares, but an installled system has a machine-id mismatch.
/etc/machine-id differes from /var/lib/dbus/machine-id, and the latter should be symlink.

**Comment by Kirek** [comment]
The machine-id warning is gone with kernel 4.2
@udeved 
Either livecd or installed system (with calamares) both machine-id match and the one in /var is a symlink to /etc/machine-id (kde edition, systemd). Doesn't matter if it's kernel 4.1 or 4.2, too.

**Comment by udeved** [comment]
Well, then systemd does something within calamares.
I had mismatch on instaled system, and the one in /var/lib/dbus was also a file.

**Comment by philmmanjaro** [comment]
@Kirek, @udeved : what exactly was tested. I know that with **aufs** it works as it should. I'll create 15.09-rc1 now with **overlayfs** to see what might happen.

**Comment by Kirek** [comment]
@philmmanjaro 
I have built the kde edition against unstable with kernel 4.1 and 4.2, and overlayfs.
With kernel 4.1 you get the warning, not in 4.2 (due to overlayfs updates).
machine-id file/symlink are fine in the livecd, and are fine in the installed system with calamares, both in 4.1 and 4.2 kernels.

**Comment by philmmanjaro** [comment]
So this concludes to use **aufs** if we use kernel 4.1 and recommend **overlayfs** from 4.2 series and up. Is the **warning** an issue btw?

**Comment by udeved** [comment]
I built an iso 5 days ago, used it to install, and had machine-id mismatch for openrc and calamares on freshly installed system.

**Comment by philmmanjaro** [comment]
The mismatch is then only **openrc** related? We have to check with **systemd**. Did you yet tried with **aufs** or **linux42**?

**Comment by udeved** [comment]
I guess phil, only for openrc and calamares.

Nope, I haven't touched 42 at all, still hunting the cgmanager annoyance..

**Comment by philmmanjaro** [comment]
That's why this is a question. If **aufs** is the way to go for the current situation I've no problem with it. However we might backport some **overlayfs** patches to **linux41**, I've to see.

**Comment by udeved** [comment]
I can only say from openrc iso side, overlayfs build works well.
I use it since 41 kernel
Also iso's initramfs seems to work better.

**Comment by Kirek** [comment]
The warning is not a big deal for a livecd, but if you want we can keep aufs for this release.
The only reason to switch to overlayfs, is the sooner we do that, the sooner you could drop the aufs patches...

**Comment by philmmanjaro** [comment]
Please pull from:

  git://git.kernel.org/pub/scm/linux/kernel/git/mszeredi/vfs.git overlayfs-next

This relaxes the requirements on the lower layer filesystem: now ones that
implement .d_revalidate, such as NFS, can be used.  Upper layer filesystems
still has the "no .d_revalidate" requirement.  Also a bad interaction with jffs2
locking has been fixed.

Thanks,
Miklos

---
Miklos Szeredi (3):
      ovl: don't traverse automount points
      ovl: allow distributed fs as lower layer
      ovl: lookup whiteouts outside iterate_dir()

I'll try to pull those patches into the next **linux41**

**Comment by Kirek** [comment]
@udeved 
Maybe calamares needs a custom machineid.conf for openrc
https://github.com/calamares/calamares/tree/master/src/modules/machineid

**Comment by Kirek** [comment]
With the latest patches the machine-id failure is gone.

**Comment by philmmanjaro** [comment]
James just reported the following: UEFI boot option fails with big blue screen "Failed to install override
security policy (14): not found" when selected from USB. Did we checked that yet?

**Comment by udeved** [comment]
I know this message, something secure boot related, my uefi bios doesn't have.

**Comment by AJSlye** [comment]
This was actually reported to me by Clemens originally.
The thing is, this doesn't seem to happen when aufs is used for the ISO.

**Comment by philmmanjaro** [comment]
I've to check that later. Currently I was playing Tomb Raider via Wine/Steam. As I don't have UEFI hardware I can only use the VirtualBox emulator. However, security options should been turned off as we don't have any signed bootloader to match those needs.

**Comment by udeved** [comment]
Regarding that message, a solution is supposedly to disable "secure boot" in uefi bios.
However, as said, I don't have such option, perhaps I had to update uefi firmware.

**Comment by AJSlye** [comment]
Still, with the two extra error messages that come up in the text console when using overlayfs as opposed to aufs, I just can't justify switching yet.
I'm already having to try and explain these two issues as it is:
Excerpt from http://wiki.netrunnerlinux.com/index.php/Netrunner_Rolling_2015.09#pre5c

BOOT from LIVE usb:
startup process halted due to:
switch_root: failed to unlink ro_branch
"Rebuild dynamic linker cache" takes up 30 seconds to initiliaze while booting from LIVE USB

**Comment by udeved** [comment]
switch_root: failed to unlink ro_branch

Is gone with overlayfs for me. Only occurs with aufs here.

Linker cache rebuild, I have never seen this.
Does it occur when systemd boots up, or is it in initramfs?

**Comment by AJSlye** [comment]
I believe that switch_root: failed to unlink ro_branch is still present on the Manjaro KDE 15.09-rc1 ISO.
I'll need to look at it again though to be sure. I'll post all the errors I see when booting this ISO.

**Comment by philmmanjaro** [comment]
machine-id message is now fixed in rc2. **switch_root** I've to check.

**Comment by philmmanjaro** [comment]
**switch_root: failed to unlink ro_branch** doesn't matter in regard of this topic. If still present in **15.09-rc2** please open a ticket.
